### PR TITLE
feat(highlights): add fzf-lua

### PR DESCRIPTION
**Pull request by strash**
[https://github.com/ibhagwan/fzf-lua](https://github.com/ibhagwan/fzf-lua)

Before
Screenshot 2024-11-26 at 04 10 08

After
Screenshot 2024-11-26 at 04 11 44

Before
Screenshot 2024-11-26 at 04 11 00

After
Screenshot 2024-11-26 at 04 12 34

### CODE REVIEW

**Comment by strash** [comment]
We should probably mention in the README about this setting for generating highlights for fzf.
```lua
require("fzf-lua").setup({ fzf_colors = true })
```

I've been using this plugin for a while now, but I just learned about this setting recently. 😀

**Comment by strash** [comment]
Some updates.

Before 
Screenshot 2024-11-26 at 14 28 43

After
Screenshot 2024-11-26 at 14 27 36

Before
Screenshot 2024-11-26 at 14 29 25

After
Screenshot 2024-11-26 at 14 28 07

**Comment by strash** [comment]
Hey @ramojus, this PR has all the changes from the PR with blink.

**Comment by ramojus** [comment]
Hey, sorry for the delays, I'm quite busy lately.

I'm just wondering if we really need to support this plugin? We have telescope, which is much more popular. Does this plugin have anything that telescope doesn't?
Otherwise the PR looks good, I would have a few minor comments and it can be merged.

**Comment by ramojus** [comment]
Also just saw that this plugin has a [`telescope` profile](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#profiles). In terms of highlight groups, with your changes, does it look similar to telescope when that profile is applied?

**Comment by strash** [comment]
> I'm just wondering if we really need to support this plugin?

[Why Fzf-Lua](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#why-fzf-lua)
Personally, I think it's a matter of personal preference, but I find it slightly faster. I'm an Fzf user, and I would greatly appreciate it if this plugin were supported by this beautiful colorscheme. There's no real reason not to support it. By doing so, we can potentially attract more users.

---

> Also just saw that this plugin has a [telescope profile](https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#profiles). In terms of highlight groups, with your changes, does it look similar to telescope when that profile is applied?

It looks like this.

Screenshot 2024-12-07 at 02 06 19
Screenshot 2024-12-07 at 02 07 56

**Comment by strash** [comment]
@ramojus I've removed variables

**Comment by ramojus** [comment]
Thanks for the screenshots, that looks surprisingly good! We should make telescope look more like this :D
But of course, it's not viable unless telescope supports darker backdrop.

This is a difficult plugin as there are a lot of scenarios. I've pulled your changes to take a deeper look. There are some things I don't like about this plugin that I haven't found a way to fix:
- Current result being bold (seems unnecessary - distracting).
image
- Backdrop removed when searching colorschemes. I get it, it's for colorscheme preview, but ideally fzf background would then be lighter when searching colorschemes.
image
- Line numbers in preview having their default dark background. (This is probably fine as long as the preview has `Normal` background)
image
I was thinking of making the backgrounds lighter, but with this issue, probably better to leave it as is.

As you know this plugin better than I do, I would appreciate any help with these issues. Otherwise, not a deal breaker.

I will also submit some review comments.

**Comment by ramojus** [comment]
Nice work, really. I like that we also retain some similarity with `fzf` cli, mainly by keeping that `FzfLuaFzfPointer` visible, but also not distracting.
One more issue here:
image
The background of that title doesn't match with normal background. Not sure which highlight group sets it, as I don't get this title by default.

**Comment by strash** [comment]
> * Current result being bold (seems unnecessary - distracting).
>   image

Not sure how to fix this. I've explicitly unset bold and set nocombine, but it seems to be ignored.  

> * Backdrop removed when searching colorschemes. I get it, it's for colorscheme preview, but ideally fzf background would then be lighter when searching colorschemes.
>   image

I think it's fine. And probably there's nothing we can do about it.

> * Line numbers in preview having their default dark background. (This is probably fine as long as the preview has `Normal` background)
>   image
>   I was thinking of making the backgrounds lighter, but with this issue, probably better to leave it as is.

Fixed.

> One more issue here: 
> image
> The background of that title doesn't match with normal background. Not sure which highlight group sets it, as I don't get this title by default.

Nice catch! Fixed.

File: lua/mellifluous/highlights/plugins/fzf.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
+local M = {}
+
+function M.set(hl, colors)
+    hl.set("FzfLuaNormal", { bg = colors.dark_bg, fg = colors.fg3 })
+    hl.set("FzfLuaBorder", { bg = colors.dark_bg, fg = colors.dark_bg })
+    hl.set("FzfLuaTitle", { fg = hl.get("Normal").fg })
+    hl.set("FzfLuaPreviewNormal", { bg = hl.get("Normal").bg, fg = hl.get("Normal").fg })
+    hl.set("FzfLuaPreviewBorder", { bg = hl.get("Normal").bg, fg = hl.get("Normal").bg })
+    hl.set("FzfLuaPreviewTitle", { link = "FzfLuaTitle" })
+    hl.set("FzfLuaCursorLine", { bg = colors.bg2, fg = colors.fg2 })
```
[the remark this answers is not in the document]
Comment (strash) [reply lua/mellifluous/highlights/plugins/fzf.lua:10]: Oh, apparently, `FzfLuaCursorLine` is intended for the preview window. I was confused about the naming convention and removed it.

**Comment by strash** [comment]
Screenshot 2024-12-14 at 23 51 09
Screenshot 2024-12-14 at 23 51 51

File: lua/mellifluous/highlights/plugins/fzf.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramojus) [reply lua/mellifluous/highlights/plugins/fzf.lua]: Oh shit, you're right, it's for preview, I also thought it's for search selection. But `FzfLuaFzfCursorLine` links to this group by default. This suggestion then applies to `FzfLuaFzfCursorLine` group:
```lua
    hl.set("FzfLuaFzfCursorLine", { bg = groups.MenuButtonSelected(colors.dark_bg).bg, fg = colors.fg3 })
```

[the remark this answers is not in the document]
Comment (strash) [reply lua/mellifluous/highlights/plugins/fzf.lua]: image

**Comment by ramojus** [comment]
Thanks!

**Pull request merged**